extmod/machine_spi.c

```diff
@@ -185,9 +179,8 @@ static mp_obj_t mp_machine_soft_spi_make_new(const mp_obj_type_t *type, size_t n
     if (args[ARG_bits].u_int != 8) {
         mp_raise_ValueError(MP_ERROR_TEXT("bits must be 8"));
     }
-    if (args[ARG_firstbit].u_int != MICROPY_PY_MACHINE_SPI_MSB) {
-        mp_raise_ValueError(MP_ERROR_TEXT("firstbit must be MSB"));
-    }
+    self->spi.firstbit = args[ARG_firstbit].u_int == MICROPY_PY_MACHINE_SPI_MSB ?
+        MICROPY_PY_MACHINE_SPI_MSB : MICROPY_PY_MACHINE_SPI_LSB;
```

Comment:
I don't think there's any need to check for valid inputs like this.  Instead doing `self->spi.firstbit = args[ARG_firstbit].u_int;` is fine (that will anyway give equivalent behaviour because `mp_soft_spi_transfer` is validating the value as well).

Alternatively, could replace the firstbit variable here with `bool msb` and do:
```c
self->spi.msb = args[ARG_firstbit].u_int == MICROPY_PY_MACHINE_SPI_MSB;
```
Then in `mp_soft_spi_transfer` it's simply `self->spi.msb ? src[i] : swap_bits(src[i])`.  That may save a few bytes of code size.

Reply:
Changed. The first variant is kept, such that the name of the symbol matches the name of the option. Since the value of MICROPY_PY_MACHINE_SPI_MSB is 0 and `self->spi.msb ? .....,` is actually self->spi.msb != 0 ? .....` most likely the compiler will generate the same code for both variants in mp_soft_spi_transfer.

Reply:
> The first variant is kept, such that the name of the symbol matches the name of the option.

Yes, good idea.

And I checked code size and indeed it doesn't change when using `bool msb`.  So the compiler is optimising it as we would expect it to.